stream-chat-android-client/src/main/java/io/getstream/chat/android/client/di/BaseChatModule.kt

```diff
@@ -63,8 +69,12 @@ internal open class BaseChatModule(
         return defaultSocket
     }
 
-    fun parser(): ChatParser {
-        return defaultParser
+    fun apiParser(): ChatParser {
+        return if (config.enableMoshi) moshiParser else gsonParser
+    }
+
+    fun eventParser(): ChatParser {
+        return gsonParser
```

Comment:
EventParser will be non-configurable for the first half of the Moshi integration, as we'll start with regular supporting API calls there